server/commands/list_objects.go

```diff
@@ -69,7 +103,34 @@ func (q *ListObjectsQuery) Execute(ctx context.Context, req *openfgapb.ListObjec
 			user:        req.User,
 			ctxTuples:   req.ContextualTuples,
 		}, resultsChan, errChan, resolvedChan)
-	}()
+	}
+
+	if !typesystem.RewriteContainsIntersection(relation) &&
```

Comment:
Could you please add a comment on what is happening (more specifically, why it's happening).

Reply:
Maybe tying in both the need for `typeinfo` and the current support for just `union` etc.

Reply:
Yep, I'll do that 👍